models/tshirt/include/Tshirt.h

```diff
@@ -0,0 +1,289 @@
+         *
+         * @param params
+         * @return
+         */
+        static double calc_Hwt(const tshirt_params& params)
+        {
+            // H_wt = alpha_fc * H_atm ; H_atm = P_atm / gamma ; P_atm = 101,300 [Pa] ; gamma = 9,810 [N m^-3] ('very nearly')
+            return params.alpha_fc * (101300 / 9810);
+        }
+
+        static double calc_Sfc(const tshirt_params& params, const tshirt_state& state)
+        {
+            // TODO: account for possiblity of Hwt being less than 0.5 (though initially, it looks like this will never be the case)
+            double z1 = calc_Hwt(params) - 0.5;
+            double z2 = z1 + 2;
+
+            // Note that z^( 1 - (1/b) ) / (1 - (1/b)) == b * (z^( (b - 1) / b ) / (b - 1)
+            return params.maxsmc * pow((1.0 / params.satpsi), (-1.0 / params.b)) *
+                   ((params.b * pow(z2, ((params.b - 1) / params.b)) / (params.b - 1)) -
+                    (params.b * pow(z1, ((params.b - 1) / params.b)) / (params.b - 1)));
+        }
+
+        static void init_nash_cascade_vector(vector<Nonlinear_Reservoir>& reservoirs, const tshirt_params& params,
+                const tshirt_state& state, const double activation, const double max_flow_velocity)
+        {
+            reservoirs.resize(params.nash_n);
+            for ( unsigned long i = 0; i < reservoirs.size(); ++i )
+            {
+                //construct a single outlet nonlinear reservoir
+                reservoirs[i] = Nonlinear_Reservoir(0, params.Ssmax, state.Snash[i], params.Kn, 1, activation,
+                        max_flow_velocity);
+            }
+        }
+
+        //! run one time step of tshirt
+        static int run(
+                double dt,
+                tshirt_params params,        //!< static parameters for tshirt
+                tshirt_state state,          //!< model state
+                tshirt_state& new_state,     //!< model state struct to hold new model state
+                tshirt_fluxes& fluxes,       //!< model flux object to hold calculated fluxes
+                double input_flux_meters,          //!< the amount water entering the system this time step
+                // TODO: should/can this be a smart pointer?
+                giuh_kernel* giuh_obj,       //!< kernel object for calculating GIUH runoff from subsurface lateral flow
+                void* et_params)            //!< parameters for the et function
+        {
+            double column_total_soil_moisture_deficit = params.Ssmax - state.Ss;
+
+            // Note this surface runoff value has not yet performed GIUH calculations
+            double surface_runoff, subsurface_infiltration_flux;
+
+            Schaake_partitioning_scheme(dt, calc_Cschaake(params), column_total_soil_moisture_deficit, input_flux_meters,
+                    &surface_runoff, &subsurface_infiltration_flux);
+
+            double Sfc = calc_Sfc(params, state);
+
+            vector<Reservoir_Outlet> subsurface_outlets;
```

Comment:
Does this need to be recreated each time step?  Or can the outlets be defined, initialized, and reused?

Reply:
This again raises the question of how things should be separated in the design.  Should reservoirs and reservoir outlets be created in/owned by the Tshirt realization object, then just passed to a call `tshirt_kernel.run(...)`?

Reply:
Reservoirs and outlets can be defined, initialized, and reused. With my limited understanding, I would vote for created in/owned by the Tshirt realization object as I think it would make things more modular.